web/packages/teleport/src/Bots/Add/GitHubActions/ConnectGitHub.tsx

```diff
@@ -0,0 +1,359 @@
+/**
+ * Teleport
+ * Copyright (C) 2023  Gravitational, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import React, { useState } from 'react';
+import styled from 'styled-components';
+import ReactSelect from 'react-select';
+
+import Box from 'design/Box';
+import Text from 'design/Text';
+import * as Icons from 'design/Icon';
+import { StyledSelect } from 'shared/components/Select';
+import Input from 'design/Input';
+import { ButtonSecondary } from 'design/Button';
+import Validation, { Validator } from 'shared/components/Validation';
+
+import Flex from 'design/Flex';
+import Card from 'design/Card';
+import ButtonIcon from 'design/ButtonIcon';
+import FieldInput from 'shared/components/FieldInput';
+
+import Alert from 'design/Alert';
+
+import Link from 'design/Link';
+
+import cfg from 'teleport/config';
+
+import { FlowButtons } from '../Shared/FlowButtons';
+import { FlowStepProps } from '../Shared/GuidedFlow';
+
+import {
+  GITHUB_HOST,
+  RefTypeOption,
+  Rule,
+  parseRepoAddress,
+  useGitHubFlow,
+} from './useGitHubFlow';
+
+const MULTIPLE_HOSTS_ERROR =
+  'All repositories must be in the same host. Please create different bots for each host.';
+const ENTERPRISE_HOST_ERROR = () => {
+  return (
+    <Box>
+      GitHub Enterprise Server Host require Teleport Enterprise.Please use a
+      repository hosted at github.com or{' '}
+      <Link target="_blank" href="https://goteleport.com/signup/enterprise/">
+        contact us
+      </Link>
+      .
+    </Box>
+  );
+};
+
+const refTypeOptions: RefTypeOption[] = [
+  {
+    label: 'any',
+    value: '',
+  },
+  {
+    label: 'Branch',
+    value: 'branch',
+  },
+  {
+    label: 'Tag',
+    value: 'tag',
+  },
+];
+
+export function ConnectGitHub({ nextStep, prevStep }: FlowStepProps) {
+  const {
+    repoRules,
+    setRepoRules,
+    addEmptyRepoRule,
+    createBot,
+    attempt,
+    resetAttempt,
+  } = useGitHubFlow();
+  const isLoading = attempt.status === 'processing';
+
+  const [hostError, setHostError] = useState<JSX.Element | string>(null);
+
+  function handleNext(validator: Validator) {
+    // clear errors
+    setHostError(null);
+
+    if (!validator.validate()) {
+      return;
+    }
+
+    const hosts = new Set<string>();
+    repoRules.forEach(rule => {
+      const { host } = parseRepoAddress(rule.repoAddress);
```

Comment:
can't the `parseRepoAddress` throw an error?

Reply:
It does. It shouldn't throw here, since the input has a rule `requireValidRepository` which already calls `parseRepoAddress` and catches the error, so this line will only be run if `parseRepoAddress` doesn't throw for all inputs.

That being said, I think it is worth adding a try/catch block here, in case something changes in other places in the future.

Updated in https://github.com/gravitational/teleport/pull/37443/commits/02fbfb313cf9b712293d4feea5c28099a3159b9c, thank you!